nix/flake/flakeref.go

```diff
@@ -292,26 +386,35 @@ func (r Ref) String() string {
 			// messed with the parsed URL.
 			return ""
 		}
-		url.RawQuery = buildQueryString("ref", r.Ref, "rev", r.Rev, "dir", r.Dir)
+		url.RawQuery = buildQueryString(url.Query(), "ref", r.Ref, "rev", r.Rev, "dir", r.Dir)
 		return url.String()
 	case TypeGitHub:
 		if r.Owner == "" || r.Repo == "" {
 			return ""
 		}
 		url := &url.URL{
-			Scheme:   "github",
-			Opaque:   buildEscapedPath(r.Owner, r.Repo, r.Rev, r.Ref),
-			RawQuery: buildQueryString("host", r.Host, "dir", r.Dir),
+			Scheme: "github",
+			Opaque: buildEscapedPath(r.Owner, r.Repo, r.Rev, r.Ref),
+			RawQuery: buildQueryString(nil,
```

Comment:
what is `nil` for?

Reply:
It's the initial `url.Values` query. In this case it's nil because we're building the query string from scratch.